backend/src/plugins/ContextMenus/actions/ban.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  ActionRowBuilder,
+  ButtonInteraction,
+  ContextMenuCommandInteraction,
+  ModalBuilder,
+  ModalSubmitInteraction,
+  TextInputBuilder,
+  TextInputStyle,
+} from "discord.js";
+import humanizeDuration from "humanize-duration";
+import { GuildPluginData } from "knub";
+import { canActOn } from "src/pluginUtils";
+import { ModActionsPlugin } from "src/plugins/ModActions/ModActionsPlugin";
+import { logger } from "../../../logger";
+import { convertDelayStringToMS, renderUserUsername } from "../../../utils";
+import { CaseArgs } from "../../Cases/types";
+import { MODAL_TIMEOUT } from "../commands/ModMenuUserCtxCmd";
+import { ContextMenuPluginType, ModMenuActionType } from "../types";
+import { updateAction } from "./update";
+
+async function banAction(
+  pluginData: GuildPluginData<ContextMenuPluginType>,
+  duration: string | undefined,
+  reason: string | undefined,
+  evidence: string | undefined,
+  target: string,
+  interaction: ButtonInteraction | ContextMenuCommandInteraction,
+  submitInteraction: ModalSubmitInteraction,
+) {
+  const interactionToReply = interaction.isButton() ? interaction : submitInteraction;
+  const executingMember = await pluginData.guild.members.fetch(interaction.user.id);
+  const userCfg = await pluginData.config.getMatchingConfig({
+    channelId: interaction.channelId,
+    member: executingMember,
+  });
+
+  const modactions = pluginData.getPlugin(ModActionsPlugin);
+  if (!userCfg.can_use || !(await modactions.hasBanPermission(executingMember, interaction.channelId))) {
+    await interactionToReply
+      .editReply({ content: "Cannot ban: insufficient permissions", embeds: [], components: [] })
+      .catch((err) => logger.error(`Ban interaction reply failed: ${err}`));
+    return;
+  }
+
+  const targetMember = await pluginData.guild.members.fetch(target);
```

Comment:
There seems to be an issue with fetching the member here. Whenever someone leaves the server it will throw an Discord error (API: Unknown Member).

Reply:
I'll have a look at fixing it over the weekend, thanks